user/tests/wiring/no_fixture/ble_provisioning.cpp

```diff
@@ -22,6 +22,11 @@
 
 // Do not enter listening mode based on the flag
 test(LISTENING_00_DISABLE_LISTENING_MODE) {
+    // If Particle.disconnect() is not called here, the Network.listen() calls
+    // could potentially interfere with ncpClient and rendering undefined
+    // behavior for some of the following tests 
+    Particle.disconnect();
```

Comment:
I think this somewhat negates the purpose of the test and hides an underlying problem.

> Network.listen() calls could potentially interfere with ncpClient and rendering undefined behavior for some of the following tests 

This should not be happening. This call should break cloud connection and enter listening mode no matter what. If 1s timing cannot be guaranteed on some platforms due to modem behavior etc, this test needs to be refactored to wait longer and/or use system events to understand that an entry into listening mode occurred.

Reply:
Strangely, this is only reproducible on proto1 hardware. Proto2 hardware doesn't have this problem.

Reply:
Perhaps post the log from when this test runs? That can help us see what's happening in the background and perhaps also augment the test with additional log statements around various calls.

Reply:
Posted the log here 🙇  https://app.shortcut.com/particle/story/107222/tracker-m-invoke-particle-connect-inside-listening-tests-cause-blocking-behavior-with-threading-enabled#activity-109610

Reply:
> If 1s timing cannot be guaranteed on some platforms due to modem behavior etc, this test needs to be refactored to wait longer 

This helped. Added the change.

Reply:
can we perhaps just change these https://github.com/particle-iot/device-os/pull/2534/commits/7f10992f503d01215193c40375c968958c2ab675#diff-2798bda0c6b54d595cbad836b0e6bff170c69d0894c9e90b0195c2d547bf6537R32 to busy-ish loops waiting for listening mode system events or just checking `Network.listening()` up to a certain large timeout until the state change actually occurs?

Reply:
Which `state` should we monitor? `Network.listening()` is being checked where ever needed [like here](https://github.com/particle-iot/device-os/pull/2534/files#diff-2798bda0c6b54d595cbad836b0e6bff170c69d0894c9e90b0195c2d547bf6537R68) - After `Network.listening()` state is changed, previously established cloud connection restores which I believe is taking time.